05_syntax_lab/04.py

```diff
@@ -1,8 +1,5 @@
-"""
-Write a program that reads lines from the user
-until an empty line is inserted.
-After the user typed in an empty line,
-print all previously inserted lines in reverse
-order (from last to first)
-"""
-
+
+def reverse(line):
+    if line !='': reverse(raw_input())
+    print line    
+reverse(raw_input())

```

Comment:
👍  🍰 That's a very interesting recursive solution @ynonp might also want to look at it. Shall we ask for a non-recursive solution instead?

I can see three issues with this one
* In order to avoid infinite recursions, Python limits the recursion depth to 1,000
* The empty line is also printed.
* It seems to be quite wasteful to call a recursive function for this.